### PR TITLE
[Impeller] Set 'enable_vulkan_validation_layers' if --unopt and Vulkan is enabled.

### PR DESCRIPTION
**Pull request by matanlurey**
Context:

> @matanlurey:
> Would it make sense for `--unopt` to imply `--enable-vulkan-validation-layers` if `--enable-vulkan` is set?
> The reason I ask is because it does seem to imply `glGetError` checks, which (I could be wrong) is roughly similar?

> @chinmaygarde:
> Makes sense.

... so uh, here it is (let me know if we prefer anything different).

### CODE REVIEW

**Review by jonahwilliams — COMMENTED** [review]
Unlike the glGetError, to see Vulkan validation errors I believe you'll need to opt in from the AndroidManifest. Maybe we should also update context_vk.cc to also automatically enabled vulkan validation in unopt modes?

**Comment by jonahwilliams** [comment]
Breadcrumbs: https://github.com/flutter/engine/blob/main/impeller/renderer/backend/vulkan/context_vk.h#L44C10-L44C27

**Review by chinmaygarde — COMMENTED** [review]
As @jonahwilliams pointed out, if we request validation but the layers are absent, [we will terminate context setup](https://github.com/flutter/engine/blob/5ab09b3a18dee536f90762f25e10cb0c21c82bb0/impeller/renderer/backend/vulkan/capabilities_vk.cc#L65) and eventually abort shell setup.

I think requiring validation in unopt modes without other guidance on how to setup validation layers puts an onerous burden on unopt engine tinkerers who might not care about Vulkan at all.

Perhaps we can relax the restriction to just log (loudly) that validations were requested but the layers weren't found but still continue with context setup? You can change the validation log to an error log there and not return `nullopt`.

Otherwise LGTM.

**Comment by jonahwilliams** [comment]
What I mean is sort of the opposite though. It's fine to compile in validation layers, but they won't do anything without a separate step to opt in. Maybe we should make that second part automatic too?

**Comment by chinmaygarde** [comment]
> It's fine to compile in validation layers...

I believe validation layers can be added to the Vulkan loader from a separate APK too. I'm afraid having some Vulkan debugging related APK installed would enable validation on Flutter apps automatically. But perhaps I am overthinking it? Maybe we should just enable it for now and deal with it later. Having validations on with the least overhead possible is going to be super valuable in the short run.

**Comment by matanlurey** [comment]
Sounds good, thanks all.

I'm setting up a test phone so I can use AGI to make sure I'm doing this all correctly - I'll convert back from a draft PR when it's ready for review!

File: tools/gn [hunk 1]
```diff
@@ -592,6 +592,11 @@ def to_gn_args(args):
 
   if args.enable_impeller_vulkan:
     gn_args['impeller_enable_vulkan'] = True
+    # As of 2023-07-26, --unopt adds "glGetError" calls for the OpenGL backend,
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply tools/gn:595]: Sounds good, removed!

**Comment by matanlurey** [comment]
Done and verified! Woot!

Screenshot 2023-07-27 at 1 33 46 PM

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**